packages/ui/src/hooks/useUserTokens.ts

```diff
@@ -0,0 +1,70 @@
+import { paths, setParams } from '@reservoir0x/reservoir-kit-client'
+import { useReservoirClient } from '@reservoir0x/reservoir-kit-ui'
+import useSWRInfinite, { SWRInfiniteConfiguration } from 'swr/infinite'
+
+type UserTokenResponse =
+  paths['/users/{user}/tokens/v5']['get']['responses']['200']['schema']
+
+type UserTokenQuery =
+  paths['/users/{user}/tokens/v5']['get']['parameters']['query']
+
+export default function (
+  user?: string | undefined,
+  options?: UserTokenQuery | false,
+  swrOptions: SWRInfiniteConfiguration = {}
+) {
+  const client = useReservoirClient()
+
+  const { data, mutate, error, isValidating, size, setSize } =
+    useSWRInfinite<UserTokenResponse>(
+      (pageIndex, previousPageData) => {
+        // options are optional, no need to exit early if not provided
```

Comment:
I think at least a user is required, and if not passed we can probably just return null as the key (preventing SWR from making a request)